src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -224,18 +228,24 @@ public void AddTagHelperAttribute(string name, object value)
         }
 
         /// <summary>
-        /// Execute and retrieve the rendered child content asynchronously.
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
         /// </summary>
+        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
+        /// of child content; cached content will be returned.</param>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
```

Comment:
Because evaluating `IHtmlContent` is deferred until `WriteTo()` is called. In some cases however, this encoder _will_ be used e.g. when `GetContent()` is called on the buffer behind the `TextWriter` in a nested tag helper scope.

That raises another overlapping issue: Though changing this `HtmlEncoder` has little affect on other pages or views, it does affect all nested tag helpers. E.g.
``` html
  <outer comment="Calls GetChildContentAsync(NullHtmlEncoder) in its Process() method">
    <inner comment="Calls GetChildContentAsync(NullHtmlEncoder) in its Process() method">
      content
    </inner>
   </outer>
```
`InnerTagHelper` will _not_ see fully HTML-encoded body content.